packages/framework-core/test/booster-graphql-dispatcher.test.ts

```diff
@@ -228,7 +233,56 @@ describe('the `BoosterGraphQLDispatcher`', () => {
             variableValues: match(graphQLVariables),
             operationName: match.any,
           })
-          expect(config.provider.graphQL.handleResult).to.have.been.calledWithExactly(graphQLResult)
+          expect(config.provider.graphQL.handleResult).to.have.been.calledWithExactly(graphQLResult, {})
+        })
+
+        it('calls the the GraphQL engine with the passed envelope and handles the result including the `responseHeaders`', async () => {
+          const graphQLBody = 'query { a { x }}'
+          const graphQLResult = { data: 'the result' }
+          const graphQLVariables = { productId: 'productId' }
+          const graphQLEnvelope: GraphQLRequestEnvelope = {
+            requestID: random.uuid(),
+            eventType: 'MESSAGE',
+            value: {
+              query: graphQLBody,
+              variables: graphQLVariables,
+            },
+          }
+          const resolverContext: GraphQLResolverContext = {
+            requestID: graphQLEnvelope.requestID,
+            operation: {
+              query: graphQLBody,
+              variables: graphQLVariables,
+            },
+            pubSub: new NoopReadModelPubSub(),
+            storeSubscriptions: true,
+            responseHeaders: {},
+          }
+          const config = mockConfigForGraphQLEnvelope(graphQLEnvelope)
+          const dispatcher = new BoosterGraphQLDispatcher(config, logger)
+          const parseSpy = spy(gqlParser, 'parse')
+          replace(gqlValidator, 'validate', fake.returns([]))
+
+          const executeFake = fake((params: any) => {
+            // Simulates that the handler has added the `responseHeaders`
+            params.contextValue.responseHeaders['Test-Header'] = 'Test-Value'
+            return graphQLResult
+          })
+          replace(gqlExecutor, 'execute', executeFake)
+
+          await dispatcher.dispatch({})
+
+          expect(parseSpy).to.have.been.calledWithExactly(graphQLBody)
+          expect(executeFake).to.have.been.calledWithExactly({
+            schema: match.any,
+            document: match.any,
+            contextValue: match(resolverContext),
+            variableValues: match(graphQLVariables),
+            operationName: match.any,
+          })
+          expect(config.provider.graphQL.handleResult).to.have.been.calledWithExactly(graphQLResult, {
+            'Test-Header': 'Test-Value',
+          })
```

Comment:
This test checks that the whole method chain is able to keep the state of the `responseHeaders` object in both directions (while the handler is being called and after the handler have changed the headers)